src/materialized/tests/sql.rs

```diff
@@ -149,6 +149,37 @@ fn test_time() -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
+#[test]
+fn test_tail_consolidation() -> Result<(), Box<dyn Error>> {
+    ore::test::init_logging();
+
+    let config = util::Config::default().workers(2);
+    let server = util::start_server(config)?;
+    let mut client_writes = server.connect(postgres::NoTls)?;
+    let mut client_reads = server.connect(postgres::NoTls)?;
+
+    client_writes.batch_execute("CREATE TABLE t (data text)")?;
+    client_reads.batch_execute(
+        "BEGIN;
+         DECLARE c CURSOR FOR TAIL t;",
+    )?;
+
+    let mut events = vec![];
+
+    let data = format!("line {}", 42);
+    client_writes.execute(
```

Comment:
I don't know if this is overkill, but would you want to add a retraction of some sort to this test?

Reply:
I added a test that outputs negative diffs/multiplicities. 👍